libs/governance/src/lib/proposals-hooks/use-proposal-event.ts

```diff
@@ -0,0 +1,75 @@
+import { useApolloClient, gql } from '@apollo/client';
+import { useCallback, useEffect, useRef } from 'react';
+import type {
+  ProposalEvent,
+  ProposalEventVariables,
+  ProposalEvent_busEvents_event_Proposal,
+} from './__generated__/ProposalEvent';
+import type { Subscription } from 'zen-observable-ts';
+
+export const PROPOSAL_EVENT_SUB = gql`
+  subscription ProposalEvent($partyId: ID!) {
+    busEvents(partyId: $partyId, batchSize: 0, types: [Proposal]) {
+      type
+      event {
+        ... on Proposal {
+          id
+          reference
+          state
+          rejectionReason
+          errorDetails
+        }
+      }
+    }
+  }
+`;
+
+export const useProposalEvent = () => {
+  const client = useApolloClient();
+  const subRef = useRef<Subscription | null>(null);
```

Comment:
question (non-blocking): curious, are there any particular reasons we have an overwhelming preference for useRef over useMemo? maybe i'm just more used to useMemo, and that's why it feels more readable to see it.

Reply:
For me useMemo doesn't make sense in this use case. We just need to store the subscription instance so we can ensure we stop the subscription when the component unmounts (or when the proposal data comes in) but also manually start it when the function is called. I think a useMemo implementation would be more complex and require more guards to prevent starting it on mount